packages/image_picker/image_picker_windows/example/lib/main.dart

```diff
@@ -0,0 +1,405 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+import 'package:image_picker_windows/image_picker_windows.dart';
+import 'package:video_player/video_player.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'Image Picker Demo',
+      home: MyHomePage(title: 'Image Picker Example'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({Key? key, this.title}) : super(key: key);
+
+  final String? title;
+
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  List<PickedFile>? _imageFileList;
+
+  set _imageFile(PickedFile? value) {
+    _imageFileList = value == null ? null : <PickedFile>[value];
+  }
+
+  dynamic _pickImageError;
+  bool isVideo = false;
+
+  VideoPlayerController? _controller;
+  VideoPlayerController? _toBeDisposed;
+  String? _retrieveDataError;
+
+  final ImagePickerPlatform _picker = ImagePickerWindows();
```

Comment:
I've tried doing this, but then I get an exception about the file_selector plugin being missing, during runtime. It seems like the FIleSelectorWindows plugin is not registering itself. If I add a:
```dart
FileSelectorWindows.registerWith();
```
at ImagePickerWindows.registerWith() method, then everything works, but it feels like wrong. Am I missing something here?

Reply:
There may well be a bug with Dart registration for the edge case of depending directly on a federated implementation (without the app-facing package). I'll investigate tomorrow or Friday.

Reply:
Okay, I know why this happens now, but I need to think about it more to figure out if it's actually a bug. There are some weird subtleties.

For the purposes of this PR, let's assume it's not a bug. In which case, the way to handle it is that you shouldn't be using `FileSelectorPlatform` anywhere in your implementation, but instead use a `FileSelectorWindows` instance directly.

Reply:
(Filed https://github.com/flutter/flutter/issues/99083 for the reason this happens)

Reply:
Should I just create my own FileSelectorWindows instance, or should I try to get the FileSelectorPlatform.instance first? A very low overhead since the MethodChannel itself is already a const outside the instance anyway, but I wonder how much this can impact other plugins that follow the same pattern.

Reply:
You can just make one; it should be minimal cost.

> I wonder how much this can impact other plugins that follow the same pattern

Yes, longer term I do think we'll need to fix the underlying issue. (I don't expect this to be a super common pattern though.)

Reply:
Ok, tests are passing, so this should be good now.